KeeAgent/KeeAgentUiThread.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Threading;
+using System.Windows.Forms;
+
+namespace KeeAgent
+{
+  /// <summary>
+  /// Provides a separate UI thread for interactive communication without interlocking with the MainWindow.
+  /// </summary>
+  public sealed class KeeAgentUiThread : IDisposable
+  {
+    private const string ComponentName = "KeeAgent Interactive UI";
+    private SynchronizationContext _synchronizationContext;
+    private ApplicationContext _applicationContext;
+
+    public KeeAgentUiThread()
+    {
+      var uiThread = new Thread(UiThreadMain) {
+        Name = ComponentName,
+        IsBackground = true // active background thread will not block the application shutdown process
+      };
+      uiThread.SetApartmentState(ApartmentState.STA);
+      uiThread.Start();
+    }
+
+    private void UiThreadMain()
+    {
+      _synchronizationContext = new WindowsFormsSynchronizationContext();
+      SynchronizationContext.SetSynchronizationContext(_synchronizationContext);
+      _applicationContext = new ApplicationContext();
+      Application.Run(_applicationContext);
```

Comment:
Do we still need the `ApplicationContext` or can we just call `Application.Run()` and use the implicit one so we don't have to keep track of it?  (see other comment below)